src/Microsoft.Health.Dicom.Blob/Features/Health/DicomConnectedStoreHealthCheck.cs

```diff
@@ -51,26 +52,33 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         _logger.LogInformation("Checking the health of the connected store.");
 
         BlobContainerClient containerClient = _blobClient.BlobContainerClient;
-        BlockBlobClient blockBlobClient = containerClient.GetBlockBlobClient($"{_externalStoreOptions.StorageDirectory}{_externalStoreOptions.HealthCheckFileName}");
 
-        // start trying to delete the blob, in case delete failed on the previous run
-        await TryDeleteBlob(blockBlobClient, cancellationToken);
+        BlockBlobClient healthCheckBlobClient = containerClient.GetBlockBlobClient($"{_externalStoreOptions.StorageDirectory}{_externalStoreOptions.HealthCheckFileName}");
+        BlockBlobClient leaseBlobClient = containerClient.GetBlockBlobClient($"{_externalStoreOptions.StorageDirectory}{_externalStoreOptions.HealthCheckLeaseFileName}");
+        BlobLeaseClient leaseBlobLeaseClient = leaseBlobClient.GetBlobLeaseClient();
 
         try
         {
-            using Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(_testContent));
+            if (!await leaseBlobClient.ExistsAsync(cancellationToken))
+            {
+                // create the blob to get a lease on if it does not exist
+                using Stream leaseStream = new MemoryStream(Encoding.UTF8.GetBytes(_leaseBlobContent));
+                await leaseBlobClient.UploadAsync(leaseStream, cancellationToken: cancellationToken);
+            }
+
+            // acquire lease on lease blob to ensure there is no conflict writing/reading/deleting the blob
+            await TryAcquireLease(leaseBlobLeaseClient, cancellationToken);
 
-            string blockId = Convert.ToBase64String(Guid.NewGuid().ToByteArray());
+            using Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(_testContent));
 
-            // test blob/write and that block blobs are supported
-            await blockBlobClient.StageBlockAsync(blockId, stream, cancellationToken: cancellationToken);
-            await blockBlobClient.CommitBlockListAsync(new List<string> { blockId }, new CommitBlockListOptions(), cancellationToken);
+            // test blob/write
+            await healthCheckBlobClient.UploadAsync(stream, cancellationToken: cancellationToken);
```

Comment:
Storage will still throw "BlobTypeNotSupported" using UploadAsync if we are using a BlockBlobClient, so testing with these specific block blob functions is not neccessary